dict/dict0dict.cc

```diff
@@ -857,9 +844,14 @@ void dict_table_remove_from_cache(dict_table_t *table) {
     index = UT_LIST_GET_LAST(table->indexes);
   }
 
-  /* Remove table from the hash tables of tables */
-  HASH_DELETE(dict_table_t, name_hash, dict_sys->table_hash, ut_fold_string(table->name), table);
-  HASH_DELETE(dict_table_t, id_hash, dict_sys->table_id_hash, ut_uint64_fold(table->id), table);
+  if (!skip_remove_from_lru) {
+    /* Remove table from lru */
+    ut_a(dict_sys->table_lookup->contains(table->name));
+    ut_a(dict_sys->table_lookup->contains(table->id));
+
+    dict_sys->table_lookup->erase(table->id);
```

Comment:
Use the lower_bound and iterator to remove the element. IIRC, it makes it an O(1) operation. You already have  lookup/contains in a runtime assertion.